File: cpp/base62.test.cpp
// charset : ASCII
#include "base62.h"
#include <stdio.h>

std::string bin2hexa(const std::vector<unsigned char>& bin)
{
	std::string ret ;
	char tmp[4] ;
	for (size_t i = 0, len = bin.size(); i < len; ++i) {
		sprintf(tmp, "%02X ", bin[i] & 0xFF) ;
		ret.append(tmp) ;
		if(i % 16 == 7)			ret.push_back(' ') ;
		else if(i % 16 == 15)	ret.push_back('\n') ;
	}
	if(bin.size() % 16 != 0)	ret.push_back('\n') ;
	return ret ;
}

int main(int argc, char *argv[])
{
	std::vector<unsigned char> bin ;
	bin.resize(256) ;
	for(size_t i = 0, len = bin.size(); i < len; ++i)
		bin[i] = (unsigned char)i ;

	printf("----bin[%zu]----\n", bin.size()) ;
	printf("%s\n", bin2hexa(bin).c_str()) ;

	std::string txt = Base62::encode(bin) ;
	printf("txt[%zu]:%s\n", txt.size(), txt.c_str()) ;

	std::vector<unsigned char> out = Base62::decode(txt) ;
	printf("----out[%zu]----\n", out.size()) ;
	printf("%s\n", bin2hexa(out).c_str()) ;

	return 0 ;
}

File: kotlin/Base62Test.kt
private fun bin2hexa(bin: ByteArray): String {
    val ret = StringBuilder()
    for (i in bin.indices) {
        ret.append(String.format("%02X ", bin[i]))
        when (i % 16) {
            7 -> ret.append(" ")
            15 -> ret.append("\n")
        }
    }
    if (bin.size % 16 != 0) ret.append("\n")
    return ret.toString()
}

fun main(args: Array<String>) {
    try {
        val bin = ByteArray(256)
        for (i in bin.indices) bin[i] = i.toByte()
        println("----bin[" + bin.size + "]----")
        println(bin2hexa(bin))
        val txt = Base62.encode(bin)
        println("txt[" + txt.length + "]:" + txt)
        val out = Base62.decode(txt)
        println("----out[" + out.size + "]----")
        println(bin2hexa(out))
    } catch (e: Exception) {
        e.printStackTrace()
    }
}